File: pmc.jl
module PMC

#=
int max_clique(long long nedges, int *ei, int *ej, int index_offset, 
                int outsize, int *clique);
=#                

const libpmc = joinpath(dirname(@Base.__FILE__),"libpmc")

function pmc(A::SparseMatrixCSC)
 maxd = Int(maximum(sum(spones(A),1)))
 ei,ej = findnz(tril(A,1))
 
 ei = map(Int32,ei)
 ej = map(Int32,ej)
 
 offset = Cint(1)
 
 outsize = maxd
 output = zeros(Int32,maxd)
 
 clique_size = ccall( 
    (:max_clique, libpmc), Cint,
    (Clonglong, Ptr{Cint}, Ptr{Cint}, Cint, Cint, Ptr{Cint}),
    length(ei), ei, ej, offset, outsize, output)
    
  return map(Int64, output[1:clique_size])
end

end

#@show PMC.pmc(sprandn(10000,10000,10/10000))

File: pmc.m
function max_clique = pmc(A)
maxd = int32(full(max(sum(spones(A),1))));
[ei,ej] = find(tril(A,1));
ei = int32(ei);
ej = int32(ej);
loadlibrary('libpmc');
offset = 1;
outsize = maxd;
output = int32(zeros(maxd,1));
eiPtr = libpointer('int32Ptr',ei);
ejPtr = libpointer('int32Ptr',ej);
outputPtr = libpointer('int32Ptr',output);
clique_size = calllib('libpmc','max_clique',size(ei,1),eiPtr,ejPtr,offset,...
                        outsize,outputPtr);
output = get(outputPtr,'Value');
max_clique = output(1:clique_size);
unloadlibrary libpmc;